### PR TITLE
✨ Split settings

### PR DESCRIPTION
**Pull request by SukiCZ**
This splits settings into a nice module 🎉 

Common variables can be configured in `settings/.env` file, while `Web3Config` has to be in python file (`local_optional.py`)
_Configuring `.env` with a large dictionary didn't seem like a good option_ 🤷🏿 

It has been tested using `python manage.py diffsettings` against local, staging & production environment ✅ 
_I have template for `.env` and `local_optional.py` for server environments_

README updated 📝 

Few things I want to test (TODO): 🗒️ 
- [x] Cloning a repository doesn't require any kind of configuration
- [ ] Checkout branch on server and compare output of `diffsettings`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
 .idea
 __pycache__
-node_modules
 static
 
 # Settings
-crowdclick/settings/local.py
-crowdclick/settings/production.py
+crowdclick/settings/.env
+crowdclick/settings/local_optional.py
```
Comment (SukiCZ) [inline .gitignore:8]: Previously ignored `local` moved to ignored `local_optional`. Current `local` contains settings with extra CORS headers for Frontend 👌🏿

File: README.md [hunk 2]
```diff
@@ -19,8 +19,9 @@ mkvirtualenv -p python3.6 -a . crowdclick
 # Install dependencies
 pip install -r requirements.txt
 
-# Initialize local settings
-echo 'from .defaults import *' > crowdclick/settings/local.py
+# (Optional) local settings
+cat crowdclick/settings/example.env > crowdclick/settings/.env
+cat crowdclick/settings/local_optional_example.py > crowdclick/settings/local_optional.py
```
Comment (SukiCZ) [inline README.md:24]: :cat: over `cp`

File: ad_source/web3_providers.py [hunk 3]
```diff
@@ -11,10 +12,11 @@
 from . import contract
 
 if typing.TYPE_CHECKING:  # pragma: no cover
-    from crowdclick.settings.defaults import web3_config_namedtuple
+    from crowdclick.settings import Web3Config
 
 
-class Web3Provider(typing.NamedTuple):
+@dataclass
```
Comment (SukiCZ) [inline ad_source/web3_providers.py:18]: `dataclass` over `namedtuple`

File: crowdclick/settings/__init__.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+env = environ.Env(
+    # SECURITY WARNING: don't run with debug turned on in production!
+    DEBUG=(bool, True),
+
+    ENV=(str, 'local'),
+
+    # SECURITY WARNING: keep the secret key used in production secret!
+    SECRET_KEY=(str, '39h^zm85$8ehrv+_1o=d-k9caiu_e=$)0oti42w9)krz4@6_0c'),
```
Comment (SukiCZ) [inline crowdclick/settings/__init__.py:30]: Note about this line - this isn't the secret you are looking for

**Review by RiccardoBiosas — APPROVED** [review]

**Pull request merged**